tensorflow/lite/experimental/shlo/ops/convolution_test.cc

```diff
@@ -0,0 +1,615 @@
+             .data = rhs_data.data()};
+  Tensor window_stride{.type = TensorType{.shape = shape_parametrs,
+                                          .element_type = DataType::kSI64},
+                       .data = window_stride_values.data()};
+  Tensor padding{.type = TensorType{.shape = shape_padding,
+                                    .element_type = DataType::kSI64},
+                 .data = padding_values.data()};
+  Tensor lhs_dilation{.type = TensorType{.shape = shape_parametrs,
+                                         .element_type = DataType::kSI64},
+                      .data = lhs_dilation_values.data()};
+  Tensor rhs_dilation{.type = TensorType{.shape = shape_parametrs,
+                                         .element_type = DataType::kSI64},
+                      .data = rhs_dilation_values.data()};
+  Tensor window_reversal{.type = TensorType{.shape = shape_parametrs,
+                                            .element_type = DataType::kI1},
+                         .data = window_reversal_values.data()};
+  Tensor inputSpatialDimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = inputSpatialDimensions_values.data()};
+  Tensor kernel_spatial_dimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = kernel_spatial_dimensions_values.data()};
+  Tensor output_spatial_dimensions{
+      .type =
+          TensorType{.shape = shape_parametrs, .element_type = DataType::kSI64},
+      .data = output_spatial_dimensions_values.data()};
+
+  Tensor output_tensor{.type = TensorType{.shape = shape_result,
+                                          .element_type = TypeParam::kStorage},
+                       .data = output_data.data()};
+
+  absl::InlinedVector<PrecisionTypes, 2> precision_configs = {
+      PrecisionTypes::DEFAULT, PrecisionTypes::DEFAULT};
+  auto op = Create(ConvolutionOp::Attributes{
+      .window_strides = window_stride,
+      .padding = padding,
+      .lhs_dilation = lhs_dilation,
+      .rhs_dilation = rhs_dilation,
+      .window_reversal = window_reversal,
+      .input_batch_dimension = input_batch_dimension,
+      .input_feature_dimension = input_feature_dimension,
+      .input_spacial_dimensions = inputSpatialDimensions,
+      .kernel_input_feature_dimension = kernel_input_feature_dimension,
+      .kernel_output_feature_dimension = kernel_output_feature_dimension,
+      .kernel_spacial_dimensions = kernel_spatial_dimensions,
+      .output_batch_dimension = output_batch_dimension,
+      .output_feature_dimension = output_feature_dimension,
+      .output_spacial_dimensions = output_spatial_dimensions,
+      .feature_group_count = feature_group_count,
+      .batch_group_count = batch_group_count,
+      .precision_configs = precision_configs});
+
+  Vector<StorageT> expected_data;
+  if (std::is_same<StorageT, BF16>::value) {
```

Comment:
These expected results seem to be differing to a larger degree than I would expect. Where were these generated from?

Reply:
we got the expected from StableHlo repo, now float types are seperated into different tests(you can check it in PR#67245)